### PR TITLE
Replace pull request https://github.com/coala/coala-bears/pull/951, which had a history beyond saving, because it was done during the pycon sprint.

### PR DESCRIPTION
**Pull request by mdorner**
Similar to java's LanguageCheck, CPPLint only supports certain kinds of
settings (use_spaces can only be true, indent_size has to be 2).

Consistent with what LanguageCheck does, this commit makes
CPPLintBear throw ValueError in case unsupported options are passed.

Extends test-case to test for exceptions being thrown for unsupported
options.

### CODE REVIEW

File: bears/c_languages/CPPLintBear.py [hunk 1]
```diff
@@ -25,13 +25,19 @@ class CPPLintBear:
...
+                         use_spaces: bool=True,
+                         indent_size: int=2,
                          cpplint_ignore: typed_list(str)=(),
                          cpplint_include: typed_list(str)=()):
         """
         :param max_line_length: Maximum number of characters for a line.
         :param cpplint_ignore:  List of checkers to ignore.
         :param cpplint_include: List of checkers to explicitly enable.
         """
+        if (not use_spaces):
```
Comment (NiklasMM) [inline bears/c_languages/CPPLintBear.py]: These parentheses are redundant.

File: bears/c_languages/CPPLintBear.py [hunk 2]
```diff
@@ -25,13 +25,19 @@ class CPPLintBear:
...
+            raise ValueError("CPPLint requires use_spaces=True")
+        if (indent_size != 2):
```
Comment (NiklasMM) [inline bears/c_languages/CPPLintBear.py]: dito. Redundant parentheses.

**Comment by sils** [comment]
hey, which issue is that related to? Can we have a `Closes <full issue url>` in the message? Also your author information seems not to be linked to your github profile - we don't care but if you want it to appear on your stats you have to set the author properly.

File: tests/c_languages/CPPLintBearTest.py [hunk 3]
```diff
@@ -13,16 +17,42 @@
...
-    CPPLintBear,
-    valid_files=(test_file,),
-    invalid_files=(),
-    settings={'cpplint_ignore': 'legal'},
-    tempfile_kwargs={"suffix": ".cpp"})
+        CPPLintBear,
+        valid_files=(test_file,),
+        invalid_files=(),
+        settings={'cpplint_ignore': 'legal'},
+        tempfile_kwargs={"suffix": ".cpp"})
```
Comment (sils) [inline tests/c_languages/CPPLintBearTest.py]: why change the indentation here? This is unrelated and we always use four spaces in those cases

File: bears/c_languages/CPPLintBear.py [hunk 4]
```diff
@@ -25,13 +25,19 @@ class CPPLintBear:
...
+                         indent_size: int=2,
                          cpplint_ignore: typed_list(str)=(),
                          cpplint_include: typed_list(str)=()):
         """
         :param max_line_length: Maximum number of characters for a line.
         :param cpplint_ignore:  List of checkers to ignore.
         :param cpplint_include: List of checkers to explicitly enable.
         """
+        if (not use_spaces):
+            raise ValueError("CPPLint requires use_spaces=True")
```
Comment (NiklasMM) [inline bears/c_languages/CPPLintBear.py]: I recently did a similar thing for `DartLintBear` (#950), there we agreed on the error message saying that the *Bear* does not support/require certain settings rather than the wrapped tool, as the user might not even know which tool they are using. So please change this to something line `CPPLintBear requires...`

File: tests/c_languages/CPPLintBearTest.py (hunk 3 above)
Comment (NiklasMM) [inline tests/c_languages/CPPLintBearTest.py]: Ẁhy did you change the indentation here?

File: tests/c_languages/CPPLintBearTest.py [hunk 5]
```diff
@@ -13,16 +17,42 @@
...
+                  'max_line_length': '13'},
+        tempfile_kwargs={"suffix": ".cpp"})
+
+
+@generate_skip_decorator(CPPLintBear)
+class CPPLintBearTest(LocalBearTestHelper):
+
+    def setUp(self):
+        self.section = Section("test section")
+        # uut = unit under test
```
Comment (sils) [inline tests/c_languages/CPPLintBearTest.py]: this is usually considered a common abbreviation. If you argue that it's not common at all we might want to change the name because we use uut in a lot of tests

File: tests/c_languages/CPPLintBearTest.py [hunk 6]
```diff
@@ -13,16 +17,42 @@
...
+        self.uut = CPPLintBear(self.section, Queue())
+
+    def test_config_failure_use_spaces(self):
+        self.section["use_spaces"] = "False"
+        self.section.append(Setting('use_spaces', "False"))
```
Comment (NiklasMM) [inline tests/c_languages/CPPLintBearTest.py]: If I'm not mistaken, you just neet the `self.section.append(Setting('use_spaces', "False"))`, not the `self.section["use_spaces"] = "False"`.
This goes for the other test cases as well

**Comment by sils** [comment]
the changes look very good and well tested!

**Comment by Mixih** [comment]
@mdorner Status? Have you made all the changes requested? if so, please rebase and update your branch so we know you are ready for re-review.

**Comment by Nosferatul** [comment]
closing due to high inactivity
